samples/nrf_desktop/src/hw_interface/motion_optical.c

```diff
@@ -847,9 +947,37 @@ static bool event_handler(const struct event_header *eh)
 		if (is_config_event(eh)) {
 			const struct config_event *event = cast_config_event(eh);
 
-			if (event->id == CONFIG_EVENT_ID_MOUSE_CPI) {
-				u16_t new_cpi = event->data[0] + (event->data[1] << 8);
-				atomic_set(&sensor_cpi, new_cpi);
+			switch (event->id) {
+			case CONFIG_EVENT_ID_MOUSE_CPI:
+				atomic_set(&sensor_cpi, (event->data[0] +
+					(event->data[1] << 8)));
+				k_sem_give(&sem);
```

Comment:
please don't semgive if we are fetching (it may break the sync as we will get additional sample)
we need to fix it is some smarter way but let's do it in another PR

Reply:
Added a simple state check. It works, sync is not broken